include/measurement_kit/nettests/ffi.hpp

```diff
@@ -96,103 +100,193 @@ class MK_NETTESTS_DEPRECATED BaseTest {
     //
     // All the following methods are straightforward because they just
     // configure the settings or register specific callbacks.
+    //
+    // XXX Make sure comments still make sense
 
     BaseTest &add_input(std::string s) {
-        impl_->settings["inputs"].push_back(std::move(s));
+        impl_.add_input(s);
         return *this;
     }
 
     BaseTest &set_input_filepath(std::string s) {
-        impl_->settings["input_filepaths"].clear();
-        return add_input_filepath(std::move(s));
+        impl_.input_filepaths.clear();
+        impl_.add_input_filepath(s);
+        return *this;
     }
 
     BaseTest &add_input_filepath(std::string s) {
-        impl_->settings["input_filepaths"].push_back(std::move(s));
+        impl_.add_input_filepath(s);
         return *this;
     }
 
     BaseTest &set_output_filepath(std::string s) {
-        impl_->settings["output_filepath"] = std::move(s);
+        impl_.set_output_filepath(s);
         return *this;
     }
 
     BaseTest &set_error_filepath(std::string s) {
-        impl_->settings["log_filepath"] = std::move(s);
+        impl_.set_log_filepath(s);
         return *this;
     }
 
     BaseTest &on_logger_eof(std::function<void()> &&fn) {
-        impl_->logger_eof_cbs.push_back(std::move(fn));
+        impl_.on_status_terminated([fn = std::move(fn)](
+                const cxx14::StatusTerminated &) noexcept {
+            MK_NETTESTS_CALL_AND_SUPPRESS(fn, ());
+        });
```

Comment:
Note to self: in theory, the `terminated` event could be emitted more than once. In practice, this would only happen when we leave the event loop and then re-enter it. However, given the circumstances, it would probably be best to somehow guarantee that the callback isn't called more than once. In fact, this callback is used by the Java bindings to free resources that are explicitly allocated, so maybe some extra _paranoia_ here is good.